shared/qt/idian/components/CheckBox.cpp

```diff
@@ -0,0 +1,24 @@
+/******************************************************************************
+    Copyright (C) 2024 by Taylor Giampaolo <warchamp7@obsproject.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include <Idian/CheckBox.hpp>
+
+using idian::CheckBox;
+
+CheckBox::CheckBox(QWidget *parent) : QCheckBox(parent), Utils(this) {}
+
+#include <Idian/moc_CheckBox.cpp>
```

Comment:
Why is this include at the bottom of the file? Does the namespace use require it?

Reply:
Having the `moc_XYZ.cpp` includes at the very bottom follows what Qt does in their own code. See e.g. https://github.com/qt/qtbase/blob/ba29d7bf99ac917b52b5d947802025751b5c6e6a/src/widgets/widgets/qabstractspinbox.cpp#L2208
It's not required, it could be moved to the top.

Reply:
I've moved them up.